### PR TITLE
[SYCL] Remove program class

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

**Review by romanovvlad — DISMISSED** [review]
LGTM

**Comment by steffenlarsen** [comment]
https://github.com/intel/llvm/pull/4667 removes the need for `use_kernel_bundle` in the submissions that use `single_task` or `parallel_for` with `kernel` parameters.

**Comment by alexbatashev** [comment]
@AlexeySachkov @Fznamznon @kbobrovs @Pennycook friendly ping

File: SYCL/KernelAndProgram/cache_env_vars_lin.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // No JITing for host devices and diffrent environment variables on linux and
 // windows.
 // REQUIRES: (level_zero || opencl) && linux
+// XFAIL: *
```
Comment (kbobrovs) [inline SYCL/KernelAndProgram/cache_env_vars_lin.cpp:6]: here and in other places - would be good to add a comment why XFAIL and what needs to happen to re-enable it

**Review by Pennycook — APPROVED** [review]

**Review by Fznamznon — APPROVED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Review by romanovvlad — APPROVED** [review]

**Comment by steffenlarsen** [comment]
https://github.com/intel/llvm/pull/4724 should make the cache work for these tests and should eliminate the need for the `XFAIL: *`.

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**